packages/dds/tree/src/simple-tree/discrepancies.ts

```diff
@@ -135,15 +158,52 @@ export interface NodeFieldsDiscrepancy {
 
 type SchemaFactoryNodeKind = "object" | "leaf" | "map";
```

Comment:
I think the name of this is very misleading, since this sounds like its referring to SchemaFactory, but actually is associating stored schema types (node kinds) with these strings. Now that this code is also doing stuff with actual schema factory node kinds, its extra confusing.

Can we replace this with 
```suggestion
type SchemaFactoryNodeKind = typeof ObjectNodeStoredSchema | typeof MapNodeStoredSchema | typeof LeafNodeStoredSchema;
```

then implement getStoredNodeSchemaType with just `nodeSchema.constructor`?

That would be so much clearer, and avoid adding a whole extra way to compare things.